resources/migrations/000_migrations.yaml

```diff
@@ -7653,3 +7653,112 @@ databaseChangeLog:
             referencedColumnNames: id
             constraintName: fk_view_log_ref_user_id
             onDelete: CASCADE
+
+# changesets 268-274 allow for handling user account emails in lowercase (GH issue 3047)
+  - property:
+      name: email_constraint_name
+      value: constraint_4
+      dbms: h2
+  - property:
+      name: email_constraint_name
+      value: core_user_email_key
+      dbms: postgresql
+  - property:
+      name: email_constraint_name
+      value: email
+      dbms: mysql,mariadb
+  - changeSet:
+      id: 268
+      author: rlotun
+      comment: Added 0.37.0   # drop unique constraint on email - this will also remove supporting indexes
+      preConditions:
+        - or:
+            - dbms:
+                type: mysql
+            - dbms:
+                type: mariadb
+            - dbms:
+                type: postgresql
+            - dbms:
+                type: h2
+      changes:
+        - dropUniqueConstraint:
+            columnDataType: varchar(254)
+            constraintName: ${email_constraint_name}
+            tableName: core_user
+  - changeSet:
+      id: 269
+      author: rlotun
+      comment: Added 0.37.0  # set email values to lower email
+      changes:
+        - sql :
+            sql : UPDATE core_user SET email = lower(email)
```

Comment:
I also think you might want to change this to `onFail: MARK_RAN` so on the off chance someone has both `cam@metabase.com` and `Cam@metabase.com` in their DB, the migrations won't fail entirely. It's definitely not a good situation to have duplicate records like that but if their app DB was already in that state it's better to let them run migrations on upgrade than have them fail and have MB fail to start.

Ideally we'd have some logic in there to fix duplicate user records automatically but that's outside of the scope of this PR. Preventing new duplicates is the first step.

Reply:
I added a defensive update to only lowercase emails that won't cause a conflict like this (thanks for the suggestion!)